### PR TITLE
enhance splitIdentifierByCaseAndSeparators to support non-case-sensitive languages

### PR DESCRIPTION
**Pull request by zrma**
This PR includes modifications to the `splitIdentifierByCaseAndSeparators` function, aiming to extend its compatibility to non-case-sensitive languages such as Korean, Chinese, and Japanese, among others.

Key Points:
- It maintains the existing functionality for case-sensitive languages.
- It integrates support for languages lacking case distinction, ensuring accurate processing while preserving the fundamental logic of the function.

However, I faced difficulties in augmenting the test suite to include these new language scenarios. My limited experience with the current test structure hindered my ability to confidently implement new test cases for these languages.

In an attempt to modify the tests, I utilized the go-jsonschema binary, executing the following command:

```bash
./go-jsonschema --capitalization HtMl,ID,URL -p test tests/data/misc/capitalization/capitalization.json -o tests/data/misc/capitalization/capitalization.go
```

This process resulted in struct names being generated as CapitalizationJson instead of Capitalization. Consequently, I manually adjusted the names.

I am uncertain about the correctness of these steps and would greatly appreciate any advice or guidance. Your feedback or suggestions for improving this feature expansion and the addition of test cases would be invaluable. 

Thank you.

### CODE REVIEW

**Comment by omissis** [comment]
hi @zrma , thanks a lot for the contribution! I will try to take a look at it over the xmas/nye holidays and hopefully merge it! 🙏

File: internal/x/text/cases.go [hunk 1]
```diff
@@ -77,39 +86,43 @@ func splitIdentifierByCaseAndSeparators(s string) []string {
 		stateNothing state = iota
 		stateLower
 		stateUpper
+		stateNonCase
```
Comment (omissis) [inline internal/x/text/cases.go]:
```suggestion
		stateNoCase
```

File: internal/x/text/cases.go [hunk 2]
```diff
@@ -77,39 +86,43 @@ func splitIdentifierByCaseAndSeparators(s string) []string {
...
-		case unicode.IsNumber(c):
+		case unicode.IsNumber(r):
 			nextState = stateNumber
 
-		default:
+		case !unicode.IsLetter(r): // Non-letter characters.
 			nextState = stateDelimiter
+
+		default: // Non-case sensitive letters.
+			nextState = stateNonCase
```
Comment (omissis) [inline internal/x/text/cases.go]:
```suggestion
			nextState = stateNoCase
```

File: internal/x/text/cases.go [hunk 3]
```diff
@@ -45,13 +45,20 @@ func (c *Caser) Identifierize(s string) string {
...
+	if len(rIdent) > 0 {
+		if !unicode.IsLetter(rIdent[0]) || isNoneCaseSensitiveLetter(rIdent[0]) {
+			ident = "A" + ident
+		}
 	}
 
 	return ident
 }
 
+func isNoneCaseSensitiveLetter(r rune) bool {
```
Comment (omissis) [inline internal/x/text/cases.go]:
```suggestion
func isNotCaseSensitiveLetter(r rune) bool {
```

File: internal/x/text/cases.go [hunk 4]
```diff
@@ -45,13 +45,20 @@ func (c *Caser) Identifierize(s string) string {
 
 	ident := sb.String()
 
-	if !unicode.IsLetter(rune(ident[0])) {
-		ident = "A" + ident
+	rIdent := []rune(ident)
+	if len(rIdent) > 0 {
+		if !unicode.IsLetter(rIdent[0]) || isNoneCaseSensitiveLetter(rIdent[0]) {
```
Comment (omissis) [inline internal/x/text/cases.go]:
```suggestion
		if !unicode.IsLetter(rIdent[0]) || isNotCaseSensitiveLetter(rIdent[0]) {
```

**Review by omissis — CHANGES_REQUESTED** [review]
Hi, thanks again for the contribution and for the patience in waiting for the review! 🙏 
The code looks good to me: if you could just push those little naming fixes I added to the PR I can then merge it! 🥂
Unfortunately I have no knowledge of non-case-sensitive languages, but if any user who does will find any bug we could fix it when the need arises. ✌️

**Comment by zrma** [comment]
> Hi, thanks again for the contribution and for the patience in waiting for the review! 🙏 The code looks good to me: if you could just push those little naming fixes I added to the PR I can then merge it! 🥂 Unfortunately I have no knowledge of non-case-sensitive languages, but if any user who does will find any bug we could fix it when the need arises. ✌️

I've made the suggested naming changes as per your review. 
Thank you for the valuable feedback and guidance! 😻  The updated commit has been pushed to the PR.

**Review by omissis — APPROVED** [review]
LGTM

**Pull request merged**